src/Container/NotFoundMiddlewareFactory.php

```diff
@@ -10,8 +10,8 @@
 namespace Zend\Expressive\Container;
 
 use Psr\Container\ContainerInterface;
-use Psr\Http\Message\ResponseInterface;
 use Zend\Expressive\Middleware\NotFoundMiddleware;
+use Zend\Expressive\NotFoundResponseInterface;
```

Comment:
Should be `Zend\Expressive\Response\NotFoundResponseInterface` (or the name I proposed; main problem is namespace).